src/Driver/AmazonSqsDriver.php

```diff
@@ -101,28 +103,38 @@ public function send(MessageInterface $message): bool
     public function wait(Closure $callback): void
     {
         while (true) {
-            $result = $this->client->receiveMessage(array(
+            if (!$this->shouldProcessNext()) {
+                break;
+            }
+            $this->checkRestart();
+
+            $result = $this->client->receiveMessage([
                 'QueueUrl' => $this->queueUrl,
-                'WaitTimeSeconds' => 20,
-            ));
+                'WaitTimeSeconds' => 1,
+            ]);
+
+//            var_dump($result);
 
             $messages = $result['Messages'];
 
             if ($messages) {
                 $hermesMessages = [];
                 foreach ($messages as $message) {
-                    $this->client->deleteMessage(array(
+//                    var_dump($message);
+                    $this->client->deleteMessage([
                         'QueueUrl' => $this->queueUrl,
                         'ReceiptHandle' => $message['ReceiptHandle'],
-                    ));
+                    ]);
                     $hermesMessages[] = $this->serializer->unserialize($message['Body']);
                 }
                 foreach ($hermesMessages as $hermesMessage) {
                     $callback($hermesMessage);
+                    $this->incrementProcessedItems();
                 }
             }
 
             if ($this->sleepInterval) {
+                $this->checkRestart();
```

Comment:
Is this second restart intentional? You have one in the beginning and one here before sleep. I would probably keep first one only. One restart check per "while cycle" is (I think) enough.

```
checkRestart()
// process message
checkRestart()
sleep()

checkRestart()
// process message
checkRestart()
sleep()

checkRestart()
// process message
checkRestart()
sleep()
```

Reply:
this is done this way by purpose. i've put `$this->checkRestart();` before _sleep()_ which i think could be useful in situation when `// process message` took significant time and you can also have `sleepInterval` set to higher number.